lib/hawkular/inventory/inventory_api_v4.rb

```diff
@@ -37,29 +37,29 @@ def self.create(hash)
     # Get single resource by id
     # @return Resource the resource
     def resource(id)
-      hash = http_get("/resources/#{id}")
+      hash = http_get("/resources/#{URI.escape(id)}")
```

Comment:
I think http_get should escape the URI directly, instead of adding it on each call.

Reply:
that would be a big refactor, as `http_get` is used in other subclasses and some of them perform a non-standard escaping that couldn't be combined with URI.escape (see https://github.com/hawkular/hawkular-client-ruby/blob/master/lib/hawkular/client_utils.rb#L20 )

Reply:
Agree, lets review that on a separate PR